src/main/java/org/reactome/release/dataexport/configuration/ConfigurationManager.java

```diff
@@ -0,0 +1,274 @@
+	 * @see ConfigurationEntryCollection
+	 */
+	ConfigurationEntryCollection getConfigurationEntryCollection() {
+		return this.configurationEntryCollection;
+	}
+
+	private void initializeConfigurationEntryCreator() {
+		this.configurationEntryCollection = new ConfigurationEntryCollection();
+	}
+
+	private String getConfigFileName() {
+		return this.configFileName;
+	}
+
+	private void addNeo4jDatabaseConfigurationEntries() {
+		configurationEntryCollection.addRequiredConfigurationEntry(
+			"Neo4j Username", "neo4jUserName"
+		);
+		configurationEntryCollection.addPasswordConfigurationEntry(
+			"Neo4j Password", "neo4jPassword"
+		);
+		configurationEntryCollection.addOptionalConfigurationEntry(
+			"Neo4j Host Server Name", "neo4jHostName", "localhost"
+		);
+		configurationEntryCollection.addOptionalConfigurationEntry(
+			"Neo4j Bolt Port", "neo4jPort", "7687"
+		);
+	}
+
+	private void addReactomeSpecificConfigurationEntries() {
+		configurationEntryCollection.addRequiredConfigurationEntry(
+			"Reactome Version", "reactomeVersion"
+		);
+		configurationEntryCollection.addOptionalConfigurationEntry(
+			"Path to Local File Output Directory", "outputDir", "output"
+		);
+	}
+
+	private void addEuropePMCFTPServerConfigurationEntries() {
+		configurationEntryCollection.addOptionalConfigurationEntry(
+			"EuropePMC FTP Server Username", "europePMCFTPUserName", "elinks"
+		);
+		configurationEntryCollection.addPasswordConfigurationEntry(
+			"EuropePMC FTP Server Password", "europePMCFTPPassword")
+		;
+		configurationEntryCollection.addRequiredConfigurationEntry(
+			"EuropePMC FTP Server Hostname URL", "europePMCFTPHostName"
+		);
+		configurationEntryCollection.addRequiredConfigurationEntry(
+			"Path to the Reactome folder on the EuropePMC FTP Server", "europePMCFTPReactomeFolderPath"
+		);
+	}
+
+	private void addNCBIFTPServerConfigurationEntries() {
+		configurationEntryCollection.addOptionalConfigurationEntry(
+			"NCBI FTP Server Username", "ncbiFTPUserName", "reactome"
+		);
+		configurationEntryCollection.addPasswordConfigurationEntry(
+			"NCBI FTP Server Password", "ncbiFTPPassword"
+		);
+		configurationEntryCollection.addRequiredConfigurationEntry(
+			"NCBI FTP Server Hostname URL", "ncbiFTPHostName"
+		);
+		configurationEntryCollection.addRequiredConfigurationEntry(
+			"Path to the Reactome folder on the NCBI FTP Server", "ncbiFTPReactomeFolderPath"
+		);
+	}
+
+	private boolean configurationFileExists() {
+		return new File(getConfigFileName()).exists();
+	}
+
+	private boolean configurationFileExistsAndIsValid() throws IOException {
+		return configurationFileExists() && configurationFileIsValid();
+	}
+
+	private String getConfigurationKeyFromConfigurationEntry(String configurationEntry) {
+		return configurationEntry.split("=")[0];
+	}
+
+	private void makeFileReadAndWriteForUserAndGroupOnly() throws IOException {
+		Runtime.getRuntime().exec("chmod 660 " + getConfigFileName());
```

Comment:
This needs to be done in a less OS-specific way. See: https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#setPosixFilePermissions(java.nio.file.Path,%20java.util.Set)

Reply:
Yeah, you're right.  I'll look into it.

Reply:
Done: https://github.com/reactome/release-data-exporter/pull/4/commits/6abb0c0ca05a69614e4473be924ba596f27cc8ba